app/controllers/api/actions_controller.rb

```diff
@@ -1,7 +1,12 @@
 class Api::ActionsController < ApplicationController
   def create
-    action = Action.create_action(action_params)
-    render json: action
+    validator = ActionValidator.new(action_params)
+    if validator.valid?
+      action = Action.create_action(action_params)
```

Comment:
`Action.create_action` actually defers to `ManageAction`, which is currently defined beneath the `Action` class. I should have moved it to `app/services`. What do you think about having `ManageAction` use your `ActionValidator`?